### PR TITLE
feature/process-special-event-workflow

### LINKED ISSUE #46: Add Process Special Event Workflow
### Feature Description

_A clear and concise description of the feature you're requesting._

Ability to process a file / video entirely using GitHub Actions.

### Use Case

_Please provide a use case to help us understand your request in context._

No need to install or setup cdp-backend or other credentials, anyone with admin creds to the repo actions could trigger a job to process a file (special event, forum, debate, etc.) this would make it easy for multiple people to help add events to the system.

### Solution

_Please describe your ideal solution._

@isaacna added a script to `cdp-backend` that allows for processing a local file (debate, forum, special event, etc) and storing to infrastructure.

Add a GitHub Action that utilizes the bin script but also downloads the remote file to the local runner before hand. As @isaacna's script assumes the file is local to begin with iirc.

Similar to other scripts, for the job to be available for use it must also have a dummy "run on push" trigger just to make it available in the GitHub UI.

### Alternatives

_Please describe any alternatives you've considered, even if you've dismissed them._

### PR DESCRIPTION
**Pull request by isaacna**
### Link to Relevant Issue
Resolves #46 

### Description of Changes

- We take in the event details as a JSON string (free of chars like \n or \t) and call the `process_special_event` script on it
- We'll get the config details from `event-gather-config.json` like we do for the event gather workflow
- Added `gcloud` to fix [an issue](https://github.com/CouncilDataProject/cdp-backend/issues/90) where the google credentials aren't implicitly being picked up. This can be seen in this [workflow run](https://github.com/isaacna/test-process-script/runs/3580510675?check_suite_focus=true). I'll fix this on cdp-backend side as well but I think using gcloud in general is good
 - I did a similar thing for [FireO](https://github.com/octabytes/FireO/pull/124) which helped manage auth easier 

### Testing
- I've run this on a test repo I've generated. ([Link to run](https://github.com/isaacna/test-process-script/runs/3581382948?check_suite_focus=true)) ([JSON string input used](https://pastebin.com/yBPnURDs))

### CODE REVIEW

File: {{ cookiecutter.hosting_github_repo_name }}/.github/workflows/process-special-event.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      run: |
+        cd python/
+        pip install .
+
+    - name: Setup gcloud
+      uses: google-github-actions/setup-gcloud@master
+      with:
+        project_id: {{ cookiecutter.infrastructure_slug }}
+        service_account_key: {% raw %}${{ secrets.GOOGLE_CREDENTIALS }}{% endraw %}
+        export_default_credentials: true
```
Comment (JacksonMaxfield) [inline {{ cookiecutter.hosting_github_repo_name }}/.github/workflows/process-special-event.yml:39]: Can you make this change to the other pipeline workflows as well.

File: {{ cookiecutter.hosting_github_repo_name }}/python/setup.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
 ]
 
 requirements = [
-    "cdp-backend[pipeline]~=3.0.0.dev11",
+    "cdp-backend[pipeline]~=3.0.0.dev13",
```
Comment (JacksonMaxfield) [inline {{ cookiecutter.hosting_github_repo_name }}/python/setup.py:20]: For safety, lets bump to dev14.

**Review by JacksonMaxfield — APPROVED** [review]
Looks good to me. Can you make those gcloud setup calls on the other pipeline workflows too?

Excited! This is such a nice feature!

File: {{ cookiecutter.hosting_github_repo_name }}/python/setup.py (hunk 2 above)
Comment (isaacna) [reply {{ cookiecutter.hosting_github_repo_name }}/python/setup.py]: Will do

File: {{ cookiecutter.hosting_github_repo_name }}/.github/workflows/process-special-event.yml (hunk 1 above)
Comment (isaacna) [reply {{ cookiecutter.hosting_github_repo_name }}/.github/workflows/process-special-event.yml:39]: Good call, added

**Pull request merged**